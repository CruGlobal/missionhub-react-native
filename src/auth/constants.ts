import { AuthPerson_currentUser_person } from './__generated__/AuthPerson';

export enum IdentityProvider {
  Apple,
  Google,
  Facebook,
  TheKey,
  Anonymous,
}

export enum AuthError {
  None,
  CredentialsIncorrect,
  EmailUnverified,
  MfaRequired,
  MfaIncorrect,
  Unknown,
}

export const emptyAuthPerson: AuthPerson_currentUser_person = {
  __typename: 'Person' as const,
  id: '',
  firstName: '',
  lastName: '',
  fullName: '',
  stage: null,
  globalRegistryMdmId: null,
  fbUid: null,
  theKeyUid: null,
};
