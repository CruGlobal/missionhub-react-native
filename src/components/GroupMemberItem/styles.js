import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  content: {
    paddingVertical: 10,
    paddingLeft: 16,
    paddingRight: 0,
  },
  detailsWrap: {
    marginTop: 1,
  },
  name: {
    fontSize: 16,
    fontWeight: 'bold',
    color: theme.primaryColor,
  },
  detailText: {
    fontSize: 14,
    color: theme.grey1,
  },
  detailTextRed: {
    fontSize: 14,
    color: theme.red,
  },
});
