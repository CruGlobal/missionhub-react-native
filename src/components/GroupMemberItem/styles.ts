import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  detailsWrap: {
    marginTop: 1,
  },
  detailText: {
    fontSize: 14,
    color: theme.grey1,
  },
  detailTextRed: {
    fontSize: 14,
    color: theme.red,
  },
});
