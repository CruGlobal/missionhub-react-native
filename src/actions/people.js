import callApi, { REQUESTS } from './api';
import { PEOPLE_WITH_ORG_SECTIONS } from '../constants';
import { findAllNonPlaceHolders } from '../utils/common';

export function getMe() {
  return (dispatch) => {
    return dispatch(callApi(REQUESTS.GET_ME));
  };
}

export function getPerson(id) {
  return (dispatch) => {
    return dispatch(callApi(REQUESTS.GET_PERSON, { person_id: id }));
  };
}

export function getPeopleList() {
  return (dispatch) => {
    const query = {
      filters: {
        assigned_tos: 'me',
      },
    };
    return dispatch(callApi(REQUESTS.GET_PEOPLE_LIST, query));
  };
}

export function getMyPeople() {
  return (dispatch, getState) => {
    const peopleQuery = {
      filters: {
        assigned_tos: 'me',
      },
      include: 'reverse_contact_assignments,organizational_permissions',
    };

    return dispatch(callApi(REQUESTS.GET_PEOPLE_LIST, peopleQuery)).then((pResults) => {
      const myPeople = findAllNonPlaceHolders(pResults, 'person'); //todo do this by looking at something other than placeholder

      const personalPeople = myPeople.filter((person) => person.organizational_permissions.length === 0);
      const personalOrg = { people: personalPeople };

      if (getState().auth.isJean) {
        return dispatch(getMinistryPeople(myPeople, personalOrg));
      } else {
        return [ personalOrg ];
      }
    });
  };
}

function getMinistryPeople(myPeople, personalOrg) {
  const orgQuery = {
    filters: {
      assigned_tos: 'me',
    },
    include: '',
  };
  return (dispatch) => {
    return dispatch(callApi(REQUESTS.GET_MY_ORGANIZATIONS, orgQuery)).then((oResults) => {
      const ministryOrgs = oResults.findAll('organization');
      const ministryPeople = myPeople.filter((person) => person.organizational_permissions.length > 0);

      ministryOrgs.forEach((org) => org.people = []);

      ministryPeople.forEach((person) => {
        person.reverse_contact_assignments.forEach((ca) => {
          const org = ministryOrgs.find((o) => {
            const foundOrg = ca.organization;

            if (foundOrg) {
              const foundOrgPerm = person.organizational_permissions.find((op) => op.organization_id === foundOrg.id);

              return o.id === foundOrg.id && foundOrgPerm;
            }
            return false;
          });
          if (org) {
            org.people.push(person);
          }
        });
      });

      return dispatch({ type: PEOPLE_WITH_ORG_SECTIONS, myOrgs: [ personalOrg, ...ministryOrgs ] });
    });
  };
}

export function searchPeople(text, filters = {}) {
  return (dispatch) => {
    if (!text) {
      return Promise.reject('NoText');
    }

    let query = {
      q: text,
      filters: {},
    };
    if (filters.ministry) {
      query.organization_ids = filters.ministry.id;
    }
    if (filters.gender) {
      query.filters.gender = filters.gender.id;
    }
    if (filters.archived) {
      query.filters.archived = true;
    }
    if (filters.unassigned) {
      query.filters.unassigned = true;
    }
    if (filters.labels) {
      query.filters.label_ids = filters.labels.id;
    }
    if (filters.groups) {
      query.filters.group_ids = filters.groups.id;
    }
    if (filters.surveys) {
      query.filters.survey_ids = filters.surveys.id;
    }

    return dispatch(callApi(REQUESTS.SEARCH, query));
  };
}