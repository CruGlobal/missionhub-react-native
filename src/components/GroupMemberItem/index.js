/* eslint complexity: 0 */

import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import i18next from 'i18next';
import { withTranslation } from 'react-i18next';

import { ORG_PERMISSIONS } from '../../constants';
import { Flex, Text, Dot, Card } from '../common';
import MemberOptionsMenu from '../MemberOptionsMenu';
import { orgPermissionSelector } from '../../selectors/people';
import { localizedStageSelector } from '../../selectors/stages';
import { orgIsUserCreated, isAdminOrOwner, isOwner } from '../../utils/common';
import ItemHeaderText from '../ItemHeaderText';

import styles from './styles';

@withTranslation('groupItem')
class GroupMemberItem extends Component {
  handleSelect = () => {
    const { onSelect, person } = this.props;
    onSelect && onSelect(person);
  };

  orgPermissionText = () => {
    const { t, personOrgPermission } = this.props;

    if (!personOrgPermission) {
      return '';
    }

    switch (`${personOrgPermission.permission_id}`) {
      case ORG_PERMISSIONS.ADMIN:
        return t('profileLabels.admin');
      case ORG_PERMISSIONS.OWNER:
        return t('profileLabels.owner');
      default:
        return '';
    }
  };

  renderUserCreatedDetails = isMe => {
    const { stagesObj, me, person } = this.props;

    let stage = null;

    const contactAssignments = person.reverse_contact_assignments || [];
    if (isMe) {
      stage = me.stage;
    } else if (stagesObj) {
      const contactAssignment = contactAssignments.find(
        a => a.assigned_to.id === me.id,
      );
      if (
        contactAssignment &&
        contactAssignment.pathway_stage_id &&
        stagesObj[`${contactAssignment.pathway_stage_id}`]
      ) {
        stage = stagesObj[`${contactAssignment.pathway_stage_id}`];
      }
    }

    const permissionText = this.orgPermissionText();

    return (
      <Fragment>
        {stage ? (
          <Text style={styles.detailText}>
            {localizedStageSelector(stage, i18next.language).name}
          </Text>
        ) : null}
        {stage && permissionText ? <Dot style={styles.detailText} /> : null}
        {permissionText ? (
          <Text style={styles.detailText}>{this.orgPermissionText()}</Text>
        ) : null}
      </Fragment>
    );
  };

  renderCruDetails = () => {
    const { t, person } = this.props;

    return (
      <Fragment>
        <Text style={styles.detailText}>
          {t('numAssigned', { count: person.contact_count || 0 })}
        </Text>
        {person.uncontacted_count ? (
          <Fragment>
            <Dot style={styles.detailText} />
            <Text style={styles.detailTextRed}>
              {t('numUncontacted', {
                count: person.uncontacted_count,
              })}
            </Text>
          </Fragment>
        ) : null}
      </Fragment>
    );
  };

  render() {
    const {
      me,
      person,
      organization,
      iAmAdmin,
      iAmOwner,
      personIsAdmin,
      personIsOwner,
      isUserCreatedOrg,
      personOrgPermission,
    } = this.props;

    const isMe = person.id === me.id;
    const showOptionsMenu = isMe || (iAmAdmin && !personIsOwner);

    return (
      <Card onPress={this.handleSelect}>
        <Flex
          value={1}
          justify="center"
          align="center"
          direction="row"
          style={styles.content}
        >
          <Flex value={1} direction="column">
            <ItemHeaderText text={person.full_name} />
            <Flex align="center" direction="row">
              {isUserCreatedOrg
                ? this.renderUserCreatedDetails(isMe)
                : this.renderCruDetails()}
            </Flex>
          </Flex>
          {showOptionsMenu ? (
            <MemberOptionsMenu
              myId={me.id}
              person={person}
              organization={organization}
              personOrgPermission={personOrgPermission}
              iAmAdmin={iAmAdmin}
              iAmOwner={iAmOwner}
              personIsAdmin={personIsAdmin}
            />
          ) : null}
        </Flex>
      </Card>
    );
  }
}

GroupMemberItem.propTypes = {
  person: PropTypes.shape({
    id: PropTypes.string.isRequired,
    full_name: PropTypes.string.isRequired,
    contact_count: PropTypes.number,
    uncontacted_count: PropTypes.number,
  }).isRequired,
  organization: PropTypes.object.isRequired,
  myOrgPermission: PropTypes.object.isRequired,
  onSelect: PropTypes.func,
};

const mapStateToProps = (
  { auth, stages },
  { person, organization, myOrgPermission },
) => {
  const personOrgPermission = orgPermissionSelector(null, {
    person,
    organization,
  });

  return {
    me: auth.person,
    stagesObj: stages.stagesObj,
    iAmAdmin: isAdminOrOwner(myOrgPermission),
    iAmOwner: isOwner(myOrgPermission),
    personIsAdmin: isAdminOrOwner(personOrgPermission),
    personIsOwner: isOwner(personOrgPermission),
    isUserCreatedOrg: orgIsUserCreated(organization),
    personOrgPermission,
  };
};

export default connect(mapStateToProps)(GroupMemberItem);
