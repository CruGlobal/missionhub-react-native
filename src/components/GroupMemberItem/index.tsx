/* eslint complexity: 0 */

import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';

import { ORG_PERMISSIONS } from '../../constants';
import { Flex, Text, Dot, Card } from '../common';
import MemberOptionsMenu from '../MemberOptionsMenu';
import { orgPermissionSelector } from '../../selectors/people';
import { orgIsUserCreated, isAdminOrOwner, isOwner } from '../../utils/common';
import ItemHeaderText from '../ItemHeaderText';

import { AuthState } from '../../reducers/auth';
import { StagesState } from '../../reducers/stages';
import { PersonProfileState } from '../../reducers/personProfile';
import { Person } from '../../reducers/people';
import { OrganizationsState } from '../../reducers/organizations';

import styles from './styles';

interface GroupMemberItemProps {
  onSelect: (person: Person) => void;
  person: Person;
  personOrgPermission: any;
  stagesObj: any;
  me: any;
  organization: any;
  iAmAdmin: boolean;
  iAmOwner: boolean;
  personIsOwner: boolean;
  personIsAdmin: boolean;
  isUserCreatedOrg: boolean;
}

const GroupMemberItem = ({
  onSelect,
  person,
  personOrgPermission,
  stagesObj,
  me,
  organization,
  iAmAdmin,
  iAmOwner,
  personIsOwner,
  personIsAdmin,
  isUserCreatedOrg,
}: GroupMemberItemProps) => {
  const { t } = useTranslation('groupItem');

  const handleSelect = () => {
    onSelect && onSelect(person);
  };

  const orgPermissionText = () => {
    if (!personOrgPermission) {
      return '';
    }

    switch (`${personOrgPermission.permission_id}`) {
      case ORG_PERMISSIONS.ADMIN:
        return t('profileLabels.admin');
      case ORG_PERMISSIONS.OWNER:
        return t('profileLabels.owner');
      default:
        return '';
    }
  };

  const renderUserCreatedDetails = (isMe: boolean) => {
    let stage = null;

    const contactAssignments = person.reverse_contact_assignments || [];
    if (isMe) {
      stage = me.stage;
    } else if (stagesObj) {
      const contactAssignment = contactAssignments.find(
        (a: any) => a.assigned_to.id === me.id,
      );
      if (
        contactAssignment &&
        contactAssignment.pathway_stage_id &&
        stagesObj[`${contactAssignment.pathway_stage_id}`]
      ) {
        stage = stagesObj[`${contactAssignment.pathway_stage_id}`];
      }
    }

    const permissionText = orgPermissionText();

    return (
      <Fragment>
        {stage ? <Text style={styles.detailText}>{stage.name}</Text> : null}
        {stage && permissionText ? <Dot style={styles.detailText} /> : null}
        {permissionText ? (
          <Text style={styles.detailText}>{orgPermissionText()}</Text>
        ) : null}
      </Fragment>
    );
  };
  // REMOVE_IN_SPLIT
  const renderCruDetails = () => {
    return (
      <Fragment>
        <Text style={styles.detailText}>
          {t('numAssigned', { count: person.contact_count || 0 })}
        </Text>
        {person.uncontacted_count ? (
          <Fragment>
            <Dot style={styles.detailText} />
            <Text style={styles.detailTextRed}>
              {t('numUncontacted', {
                count: person.uncontacted_count,
              })}
            </Text>
          </Fragment>
        ) : null}
      </Fragment>
    );
  };

  const isMe = person.id === me.id;
  const showOptionsMenu = isMe || (iAmAdmin && !personIsOwner);

  return (
    <Card onPress={handleSelect}>
      <Flex
        value={1}
        justify="center"
        align="center"
        direction="row"
        style={styles.content}
      >
        <Flex value={1} direction="column">
          <ItemHeaderText text={person.full_name} />
          <Flex align="center" direction="row">
            {isUserCreatedOrg
              ? renderUserCreatedDetails(isMe)
              : renderCruDetails()}
          </Flex>
        </Flex>
        {showOptionsMenu ? (
          <MemberOptionsMenu
            myId={me.id}
            person={person}
            organization={organization}
            personOrgPermission={personOrgPermission}
            iAmAdmin={iAmAdmin}
            iAmOwner={iAmOwner}
            personIsAdmin={personIsAdmin}
          />
        ) : null}
      </Flex>
    </Card>
  );
};

const mapStateToProps = (
  { auth, stages }: { auth: AuthState; stages: StagesState },
  {
    person,
    organization,
    myOrgPermission,
  }: {
    person: PersonProfileState;
    organization: OrganizationsState;
    myOrgPermission: any;
  },
) => {
  const personOrgPermission = orgPermissionSelector(
    {},
    {
      person,
      organization,
    },
  );

  return {
    me: auth.person,
    stagesObj: stages.stagesObj,
    iAmAdmin: isAdminOrOwner(myOrgPermission),
    iAmOwner: isOwner(myOrgPermission),
    personIsAdmin: isAdminOrOwner(personOrgPermission),
    personIsOwner: isOwner(personOrgPermission),
    isUserCreatedOrg: orgIsUserCreated(organization),
    personOrgPermission,
  };
};

export default connect(mapStateToProps)(GroupMemberItem);
