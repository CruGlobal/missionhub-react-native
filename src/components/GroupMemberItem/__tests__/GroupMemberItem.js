import React from 'react';

import { ORG_PERMISSIONS } from '../../../constants';
import { orgPermissionSelector } from '../../../selectors/people';
import {
  renderShallow,
  testSnapshotShallow,
  createMockStore,
} from '../../../../testUtils';

import GroupMemberItem from '..';

jest.mock('../../../selectors/people');

const stage = {
  id: 1,
  name: 'Uninterested',
};

const myId = '1';
const me = {
  id: myId,
  full_name: 'Me',
  stage,
};

const member = {
  id: '123',
  full_name: 'Full Name',
  contact_count: 5,
  uncontacted_count: 3,
};

const orgPermission = { id: '111' };

const memberPermissions = {
  ...orgPermission,
  permission_id: ORG_PERMISSIONS.USER,
};
const adminPermissions = {
  ...orgPermission,
  permission_id: ORG_PERMISSIONS.ADMIN,
};
const ownerPermissions = {
  ...orgPermission,
  permission_id: ORG_PERMISSIONS.OWNER,
};

const organization = { id: '1234', user_created: false };

const reverse_contact_assignments = [
  { id: '555', assigned_to: { id: myId }, pathway_stage_id: stage.id },
];

const props = {
  onSelect: jest.mock(),
  person: member,
  myOrgPermission: orgPermission,
  myId,
  organization,
};

const store = createMockStore({
  auth: {
    person: me,
  },
  stages: {
    stagesObj: {
      [`${stage.id}`]: stage,
    },
  },
});

beforeEach(() => {
  orgPermissionSelector.mockReturnValue(memberPermissions);
});

describe('render contacts count', () => {
  describe('user created org', () => {
    const newOrg = { ...organization, user_created: true };

    it('should render no stage, member permissions', () => {
      testSnapshotShallow(
        <GroupMemberItem {...{ ...props, organization: newOrg }} />,
        store,
      );
    });

    it('should render no stage, admin permissions', () => {
      orgPermissionSelector.mockReturnValue(adminPermissions);
      testSnapshotShallow(
        <GroupMemberItem {...{ ...props, organization: newOrg }} />,
        store,
      );
    });

    it('should render no stage, owner permissions', () => {
      orgPermissionSelector.mockReturnValue(ownerPermissions);
      testSnapshotShallow(
        <GroupMemberItem {...{ ...props, organization: newOrg }} />,
        store,
      );
    });

    it('should render stage, member permissions', () => {
      const newMember = {
        ...member,
        reverse_contact_assignments,
      };
      testSnapshotShallow(
        <GroupMemberItem
          {...{ ...props, organization: newOrg, person: newMember }}
        />,
        store,
      );
    });

    it('should render stage, admin permissions', () => {
      const newMember = {
        ...member,
        reverse_contact_assignments,
      };
      orgPermissionSelector.mockReturnValue(adminPermissions);
      testSnapshotShallow(
        <GroupMemberItem
          {...{ ...props, organization: newOrg, person: newMember }}
        />,
        store,
      );
    });

    it('should render stage, owner permissions', () => {
      const newMember = {
        ...member,
        reverse_contact_assignments,
      };
      orgPermissionSelector.mockReturnValue(ownerPermissions);
      testSnapshotShallow(
        <GroupMemberItem
          {...{ ...props, organization: newOrg, person: newMember }}
        />,
        store,
      );
    });
  });

  describe('cru org', () => {
    it('should render assigned and uncontacted', () => {
      testSnapshotShallow(<GroupMemberItem {...props} />, store);
    });

    it('should render 0 assigned', () => {
      const newMember = { ...member, contact_count: 0 };
      testSnapshotShallow(
        <GroupMemberItem {...{ ...props, person: newMember }} />,
        store,
      );
    });

    it('should render 0 uncontacted', () => {
      const newMember = { ...member, uncontacted_count: 0 };
      testSnapshotShallow(
        <GroupMemberItem {...{ ...props, person: newMember }} />,
        store,
      );
    });
  });
});

describe('render MemberOptionsMenu', () => {
  it('should render menu if person is me', () => {
    orgPermissionSelector.mockReturnValue(memberPermissions);
    const newMember = { ...member, id: myId };
    testSnapshotShallow(
      <GroupMemberItem {...{ ...props, person: newMember }} />,
      store,
    );
  });

  it('should render menu if I am admin and person is not owner', () => {
    orgPermissionSelector.mockReturnValue(memberPermissions);
    testSnapshotShallow(
      <GroupMemberItem {...{ ...props, myOrgPermission: adminPermissions }} />,
      store,
    );
  });

  it('should not render menu if person is owner', () => {
    orgPermissionSelector.mockReturnValue(ownerPermissions);
    testSnapshotShallow(
      <GroupMemberItem {...{ ...props, myOrgPermission: adminPermissions }} />,
      store,
    );
  });

  it('should not render menu if I am member', () => {
    orgPermissionSelector.mockReturnValue(memberPermissions);
    testSnapshotShallow(<GroupMemberItem {...props} />, store);
  });
});

describe('onSelect', () => {
  it('calls onSelect prop', () => {
    const onSelect = jest.fn();

    renderShallow(<GroupMemberItem {...{ ...props, onSelect }} />, store)
      .instance()
      .handleSelect();

    expect(onSelect).toHaveBeenCalled();
  });
});
