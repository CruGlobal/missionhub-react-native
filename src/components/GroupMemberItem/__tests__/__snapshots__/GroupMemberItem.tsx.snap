// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MemberOptionsMenu should not render menu if I am member 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          5 Assigned
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#FF5532",
              "fontSize": 14,
            }
          }
        >
          3 Uncontacted
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render MemberOptionsMenu should not render menu if person is owner 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          5 Assigned
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#FF5532",
              "fontSize": 14,
            }
          }
        >
          3 Uncontacted
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render MemberOptionsMenu should render menu if I am admin and person is not owner 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          5 Assigned
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#FF5532",
              "fontSize": 14,
            }
          }
        >
          3 Uncontacted
        </MyText>
      </Flex>
    </Flex>
    <Connect(withI18nextTranslation(MemberOptionsMenu))
      iAmAdmin={true}
      iAmOwner={false}
      myId="1"
      organization={
        Object {
          "id": "1234",
          "user_created": false,
        }
      }
      person={
        Object {
          "contact_count": 5,
          "full_name": "Full Name",
          "id": "123",
          "uncontacted_count": 3,
        }
      }
      personIsAdmin={false}
      personOrgPermission={
        Object {
          "id": "111",
          "permission_id": "4",
        }
      }
    />
  </Flex>
</Card>
`;

exports[`render MemberOptionsMenu should render menu if person is me 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          5 Assigned
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#FF5532",
              "fontSize": 14,
            }
          }
        >
          3 Uncontacted
        </MyText>
      </Flex>
    </Flex>
    <Connect(withI18nextTranslation(MemberOptionsMenu))
      iAmAdmin={false}
      iAmOwner={false}
      myId="1"
      organization={
        Object {
          "id": "1234",
          "user_created": false,
        }
      }
      person={
        Object {
          "contact_count": 5,
          "full_name": "Full Name",
          "id": "1",
          "uncontacted_count": 3,
        }
      }
      personIsAdmin={false}
      personOrgPermission={
        Object {
          "id": "111",
          "permission_id": "4",
        }
      }
    />
  </Flex>
</Card>
`;

exports[`render contacts count cru org should render 0 assigned 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          0 Assigned
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#FF5532",
              "fontSize": 14,
            }
          }
        >
          3 Uncontacted
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count cru org should render 0 uncontacted 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          5 Assigned
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count cru org should render assigned and uncontacted 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          5 Assigned
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#FF5532",
              "fontSize": 14,
            }
          }
        >
          3 Uncontacted
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count user created org should render no stage, admin permissions 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Admin
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count user created org should render no stage, member permissions 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      />
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count user created org should render no stage, owner permissions 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Owner
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count user created org should render stage, admin permissions 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Uninterested
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Admin
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count user created org should render stage, member permissions 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Uninterested
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;

exports[`render contacts count user created org should render stage, owner permissions 1`] = `
<Card
  onPress={[Function]}
>
  <Flex
    align="center"
    direction="row"
    justify="center"
    style={
      Object {
        "height": 50,
        "paddingLeft": 16,
        "paddingRight": 0,
        "paddingVertical": 10,
      }
    }
    value={1}
  >
    <Flex
      direction="column"
      value={1}
    >
      <ItemHeaderText
        text="Full Name"
      />
      <Flex
        align="center"
        direction="row"
      >
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Uninterested
        </MyText>
        <Dot
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        />
        <MyText
          style={
            Object {
              "color": "#B2B0B2",
              "fontSize": 14,
            }
          }
        >
          Owner
        </MyText>
      </Flex>
    </Flex>
  </Flex>
</Card>
`;
